### PR TITLE
Fix SetPlayerTeam and vehicle friendly fire (#128)

### LINKED ISSUE #128: SetPlayerTeam and explosions not affecting vehicles
Hello,

So I am currently using an outdated version, maybe this was fixed by a future update, I dont know. I can try upgrading later.

But anyways, I use SetPlayerTeam on my script to set teams. When I do this players can not damage vehicles through explosions

You can try this yourself. Make a command to set a players team using SPT function. Then set yourself to team 0 and the other player to 1. Make the other player enter a car. Throw the grenade to the car or shoot at it with a tank. The car wont get damaged. 

How can this be solved?

### PR DESCRIPTION
**Pull request by AliLogic**
Fixes #128. Running this piece of code on my server since almost a week now, no issues and works completely fine.

Now using `EnableVehicleFriendlyFire();` should lets you shoot enemy vehicles and keeps your own team vehicles safe and sound.

### CODE REVIEW

File: weapon-config.inc [hunk 1]
```diff
@@ -2658,10 +2658,10 @@ public OnPlayerDeath(playerid, killerid, reason)
 			s_DeathSkip[playerid] = 2;
 
 			ForceClassSelection(playerid);
-			SetSpawnInfo(playerid, 0xFE, GetPlayerSkin(playerid), x, y, z, r, 0, 0, 0, 0, 0, 0);
+			SetSpawnInfo(playerid, /*0xFE*/s_PlayerTeam[playerid], GetPlayerSkin(playerid), x, y, z, r, 0, 0, 0, 0, 0, 0);
```
Comment (ADRFranklin) [inline weapon-config.inc:2661]: Don't push code that has commented out previous values like this. Please remove them.

File: weapon-config.inc [hunk 2]
```diff
@@ -2658,10 +2658,10 @@ public OnPlayerDeath(playerid, killerid, reason)
...
 			TogglePlayerSpectating(playerid, true);
 			TogglePlayerSpectating(playerid, false);
-			SetSpawnInfo(playerid, 0xFE, GetPlayerSkin(playerid), x, y, z, r, 0, 0, 0, 0, 0, 0);
+			SetSpawnInfo(playerid, /*0xFE*/s_PlayerTeam[playerid], GetPlayerSkin(playerid), x, y, z, r, 0, 0, 0, 0, 0, 0);
```
Comment (ADRFranklin) [inline weapon-config.inc:2664]: This one too

File: weapon-config.inc [hunk 3]
```diff
@@ -2962,7 +2962,7 @@ public OnPlayerUpdate(playerid)
 				GetPlayerPos(playerid, x, y, z);
 				GetPlayerFacingAngle(playerid, r);
 
-				SetSpawnInfo(playerid, 0xFE, GetPlayerSkin(playerid), x, y, z, r, 0, 0, 0, 0, 0, 0);
+				SetSpawnInfo(playerid, /*0xFE*/s_PlayerTeam[playerid], GetPlayerSkin(playerid), x, y, z, r, 0, 0, 0, 0, 0, 0);
```
Comment (ADRFranklin) [inline weapon-config.inc:2965]: and this one

File: weapon-config.inc [hunk 4]
```diff
@@ -4244,7 +4244,7 @@ static SpawnPlayerInPlace(playerid) {
 	GetPlayerPos(playerid, x, y, z);
 	GetPlayerFacingAngle(playerid, r);
 
-	SetSpawnInfo(playerid, 0xFE, GetPlayerSkin(playerid), x, y, z, r, 0, 0, 0, 0, 0, 0);
+	SetSpawnInfo(playerid, /*0xFE*/s_PlayerTeam[playerid], GetPlayerSkin(playerid), x, y, z, r, 0, 0, 0, 0, 0, 0);
```
Comment (ADRFranklin) [inline weapon-config.inc:4247]: Finally this one

**Comment by ADRFranklin** [comment]
I have also noticed a few issues with this PR, there are a few checks in the code that checks if the Team is `0xFE`. However you have ignored these checks from the changes.

These checks will now never be true, because it will never be `0xFE` in the same way. This has not been fully tested and I won't be merging it until I can confirm this doesn't affect current gamemodes.

**Comment by AliLogic** [comment]
I will fix those stuff and have tested it on my script running publically with ~70 players, even tested another script and this works fine.

**Comment by InterSstellar** [comment]
That Will Be A Good Update. I Was Going To Use `EnableVehicleFriendlyFire()` But I Failed.
If You Updated It And It Worked Without Issues Please Share With Us.

**Comment by AliLogic** [comment]
I have done the changes that I was running the server with for a while;  No issues.  Confirm if anything is wrong or needs explanation;  Thanks.

**Comment by ADRFranklin** [comment]
Have you tested all the features of weapon-config to make sure this doesn't effect anything, like weapon-damage, team damage, custom damage, etc?

**Comment by AliLogic** [comment]
If you look at the code, it doesn't touch that at all; and with my testing, I use custom damages, team damage stuff and that isn't impacted.  If you still feel doubtful, go test it yourself or encourage others to do it.  I have made a PR that highlights the issue, if you aren't satisfied with the code, post a reply I'll take a look at it.

**Comment by InterSstellar** [comment]
I Found A Bug In This Update And That Is When You Use SetPlayerTeam When The Player Is Already Spawned It Won't Change Player Team And Just The `s_PlayerTeam[playerid]` Will Change And This Will Cause Player Can Damage Player Vehicle In Same Team Until Respawn I Add `SetPlayerTeam` Under `stock WC_SetPlayerTeam(playerid, team)` And It Fixed The Problem.

My English Is Bad And Also My Coding Skills Are Bad I Hope You Understand What I'm Saying And I Hope I Was Right ;D

**Comment by ADRFranklin** [comment]
@AliLogic I told you to test all cases, clearly you didn't test it properly -_-

**Comment by AliLogic** [comment]
On the two scripts I tested, I didn't find this issue.  If you did, props to you.  Make a PR, thanks.

**Pull request merged**